clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -377,36 +389,41 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         bool have_cond_branch_target = false;
         addr_t cond_branch_target = 0;
         if (shard->prev_instr_.instr.addr != 0 /*first*/ &&
-            type_is_instr_direct_branch(shard->prev_instr_.instr.type) &&
-            // We do not bother to support legacy traces without encodings.
-            TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, shard->file_type_)) {
-            addr_t trace_pc = shard->prev_instr_.instr.addr;
+            type_is_instr_direct_branch(shard->prev_instr_.instr.type) && has_encoding) {
             if (shard->prev_instr_.instr.encoding_is_new)
                 shard->branch_target_cache.erase(trace_pc);
```

Comment:
In this check, we should use the prev instr's pc (prev_instr_.instr.addr). `trace_pc` erroneously contains that currently (see other comment about wanting to decode the **current** instr in each iteration, which means `trace_pc` should really have the current instr's pc).

Reply:
Reminder about this unresolved comment.